packages/pg-gateway/src/auth/cert.ts

```diff
@@ -43,52 +42,52 @@ export class CertAuthFlow extends BaseAuthFlow {
     this.username = params.username;
   }
 
-  async handleClientMessage(message: Buffer): Promise<void> {
-    if (!(this.socket instanceof TLSSocket)) {
-      this.sendError({
+  async *handleClientMessage(message: BufferSource) {
+    // biome-ignore lint/correctness/noConstantCondition: TODO: detect TLS state
+    if (false) {
+      yield createBackendErrorMessage({
         severity: 'FATAL',
         code: '08000',
         message: `ssl connection required when auth mode is 'certificate'`,
       });
-      this.socket.end();
-      return;
+      throw new Error('end socket');
     }
 
-    if (!this.socket.authorized) {
-      this.sendError({
+    // biome-ignore lint/correctness/noConstantCondition: TODO: detect if cert authorized
+    if (false) {
+      yield createBackendErrorMessage({
         severity: 'FATAL',
         code: '08000',
         message: 'client certificate is invalid',
       });
-      this.socket.end();
-      return;
+      throw new Error('end socket');
     }
 
-    this.socket.pause();
-    const isValid = await this.auth.validateCredentials(
-      {
-        username: this.username,
-        certificate: this.socket.getPeerCertificate(),
-      },
-      this.connectionState,
-    );
-    this.socket.resume();
+    // TODO: get peer cert and validate through hook
+    const isValid = false;
+
+    // const isValid = await this.auth.validateCredentials(
+    //   {
+    //     username: this.username,
+    //     certificate:  this.socket.getPeerCertificate(),
+    //   },
+    //   this.connectionState,
+    // );
```

Comment:
WIP need to figure out how to deal with certs in a cross-platform way.

Reply:
we can get the certificate from the Connection's options, no need to use `getPeerCertificate` here to me.

Reply:
But this is the client cert right? That will only be available at runtime.

Reply:
Maybe we defer validation to the platform similar to `upgradeTls`, and just pass the raw PEM binary here - the platform can decide how to parse and validate the cert.

Reply:
Or another option is to use a lib like [pkijs](https://github.com/PeculiarVentures/PKI.js) to parse/validate that is cross platform.

Reply:
> But this is the client cert right? That will only be available at runtime.

I wasn't sure but ChatGPT seems to agree that it's the server's certificate:

>  If you are on the server side and already have access to the .pem files containing the SSL/TLS certificates and private keys, you generally do not need to use getPeerCertificate.
	•	Server Side:
	•	On the server side, you’re typically the one presenting the certificate to the client during the SSL/TLS handshake. You configure your server with the necessary certificates (e.g., using .pem files) and private keys.
	•	Since you’re the one providing the certificate, you already have access to all its details. Therefore, there’s no need to use getPeerCertificate to retrieve it—this method is more relevant for clients or other scenarios where you don’t already have the certificate.
	•	Client Side:
	•	getPeerCertificate is more commonly used on the client side or in scenarios where you need to validate or inspect the certificate presented by the other party in the connection. For example, if your Node.js server is acting as a client to another server (e.g., making HTTPS requests), you might use getPeerCertificate to inspect the server’s certificate.

Reply:
It's the client cert 😃 most TLS use cases only use server side certs which is what ChatGPT is referring to, but the spec supports client side certs too as a way for the server to authenticate the client (usually an alternative to passing credentials). Postgres' `cert` auth method is exactly this, and that's what this code implements.

Reply:
Solved by deferring validation to the platform. Passes around certs as raw DER `Uin8Array` which the platform can parse and validate.